pkg/controller/dnszone/dnszone_controller.go

```diff
@@ -215,10 +217,43 @@ func (r *ReconcileDNSZone) reconcileDNSProvider(actuator Actuator, dnsZone *hive
 	r.logger.Debug("Retrieving current state")
 	err := actuator.Refresh()
 	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
```

Comment:
We also need to check the code on the error. Not every `awserr.Error` is going to be caused by insufficient creds.